test/integration/hds_integration_test.cc

```diff
@@ -396,6 +412,9 @@ TEST_P(HdsIntegrationTest, SingleEndpointHealthyTcp) {
   RELEASE_ASSERT(result, result.message());
   host_upstream_->set_allow_unexpected_disconnects(true);
 
+  // Make sure the cluster processed the endpoint's response
+  test_server_->waitForCounterGe("cluster.anna.health_check.success", 1);
```

Comment:
It's a bit unclear to me what the win with some of the waits is. We will need to wait for the gRPC response from Envoy to the HDS server below, and those messages are sent independently of what we're doing here wait wise.